Финальное задания Спринта 7

isakovleonid opened the pull request:
1) Добавлена реализация FileBackedTaskManager
2) Исправление int на Integer для реквизита Task.id 
3) Нумерация задач начинается с 1
4) Исправлены и дополнены тест-кейсы

isakovleonid committed:
Финальное задания Спринта 7

1) Добавлена реализация FileBackedTaskManager
2) Исправление int на Integer для реквизита Task.id
3) Нумерация задач начинается с 1
4) Исправлены и дополнены тест-кейсы

Alexander-Bolotov commented on src/ru/isakovleonid/practicum/taskmanager/taskmanager/Task.java:
```diff
@@ -1,7 +1,11 @@
+
+import static java.lang.String.join;
+
 public class Task {
```
Задачи и их тип удобнее будет вынести в отдельный пекедж

Alexander-Bolotov commented on src/ru/isakovleonid/practicum/taskmanager/taskmanager/FileBackedTaskManager.java:
```diff
@@ -0,0 +1,169 @@
+        Path path = Paths.get(fileName);
+        if (Files.notExists(path)) {
+            try {
+                Path newFile = Files.createFile(path);
```
переменная нигде не используется

Alexander-Bolotov commented on src/ru/isakovleonid/practicum/taskmanager/taskmanager/FileBackedTaskManager.java:
```diff
@@ -0,0 +1,169 @@
+                    cntLine++;
+                    if (cntLine > 1) {
+                        String[] taskParams = line.split(delimiter);
+
```
не обязательно, но можно добавить проверку, что каждая строка данных имеет необходимое количество столбцов (например, для задачи их минимум 5)

Alexander-Bolotov replied:
а так же что строка не пустая

isakovleonid committed:
Финальное задания Спринта 7

1) Доработки по результату code review
2) Реализован метод FileBackedTaskManager.loadFromFile
